vmm/src/api/http/http_endpoint.rs

```diff
@@ -490,11 +527,38 @@ impl PutHandler for VmRestore {
         api_notifier: EventFd,
         api_sender: Sender<ApiRequest>,
         body: &Option<Body>,
-        files: Vec<File>,
+        mut files: Vec<File>,
     ) -> std::result::Result<Option<Body>, HttpError> {
         if let Some(body) = body {
             let mut restore_cfg: RestoreConfig = serde_json::from_slice(body.raw())?;
 
+            // First, consume mem FDs, if any.
+            let expected_mem_fds = restore_cfg.mem_fds.as_ref().map_or(0, |fds| fds.len());
+
+            if files.len() < expected_mem_fds {
```

Comment:
it looks to me that you replicate logic here that is supposed to be abstracted behind `attach_fds_to_cfgs`. The new idea of that abstraction was exactly to not have these kind of statements in the handler